### DISCUSSION TITLE
web3数据开发规范

### DISCUSSION THREAD

**Discussion by zhangmaosen** [Tagee-xyz/tagee#8] [General]
先抛个砖：
1. 表的schema规范：命名等
2. 表的使用说明规范：需要包含的必要描述内容
3. 任务调度规范
4. 数据质量规范
等等

**Comment by mark800** [comment]
觉得需要先明确需求，规范是为需求服务的

**Reply by zhangmaosen** [reply]
是的，的确缺了这个东西。需求，我知道的主要是这三类：
1. 计算逻辑相关的，也就是要计算的具体问题。目前Tagee第一步，是要把描述web3项目基本面的指标都要计算出来，并且把处理逻辑都开源公开。具体要计算的指标定义，我正在整理，预计需要2天来完成。
3. Data OPS相关的：开发、测试、上线、运维相关的流程与规范，为了保障质量。比如命名规范，代码格式规范，测试流程与持续集成如何做，运维告警怎么搞，等等
4. Data Gov相关的：为了是让大家能够更好的协作，互相复用，沉淀标准，减少数据（含元数据）不一致带来的冲突和重复工作

**Comment by zhangmaosen** [comment]
没错，我来整理一版需求。😄

Mark_Yang ***@***.***>于2022年9月7日 周三上午10:51写道：

**Comment by wuguoyong** [comment]
有一些基本的规范是需要遵循的，规范需要进行分类，像软件程序编码规范、数据库开发规范和数据治理方法论落地规范等
1. 表的设计，比如逻辑建模、物理建模，这些是必要的建模过程，命名、描述这些也可以按照实际需要进行规范约束
2. 数据质量规范，可以按照数据治理方法论里的数据质量进行规范，甚至内置一些规则库，比如完整性、一致性、唯一性这些维度
